8968-rest-bundle-creation done (#8988)
magento/devdocs: src/guides/v2.3/rest/tutorials/bundle-product/create-bundle-products.md

---
group: rest-api
title: Create bundle products tutorial
functional_areas:
  - Integration
---

A bundle is a build your own, customizable product. Each bundle product can be either
[Simple Product](https://docs.magento.com/user-guide/catalog/product-create-simple.html) or [Virtual Product](https://docs.magento.com/user-guide/catalog/product-create-virtual.html)

This tutorial describes how you can use the Magento REST API to create bundle products.

{:.bs-callout-info}

> Install a Magento 2.3 (or later) instance with sample data defines a functional store, called Luma. Use REST client to send calls to Magento , see [Postman](https://www.getpostman.com/). Obtain an admin authorization token. See [How to generate the admin token]({{ page.baseurl }}/rest/tutorials/prerequisite-tasks/create-admin-token.html) for more information. [REST Tutorials]({{ page.baseurl }}/rest/tutorials/index.html) provides additional information about completing any Magento REST tutorial.

Reviewer: ```suggestion
Install a Magento 2.3 (or later) instance with sample data which creates a functional store using the Luma theme. Use REST client to send calls to Magento , see [Postman](https://www.getpostman.com/). Obtain an admin authorization token. See [How to generate the admin token]({{ page.baseurl }}/rest/tutorials/prerequisite-tasks/create-admin-token.html) for more information. [REST Tutorials]({{ page.baseurl }}/rest/tutorials/index.html) provides additional information about completing any Magento REST tutorial.
```

Author: as a configurable product, I will change & push the changes @BarnyShergold

Author: Above grammatical changes updated @BarnyShergold

             "value": "New bundle"
          },
          {
             "attribute_code": "sku_type",
             "value": "0"
          },
          {
             "attribute_code": "meta_keyword",
             "value": "New bundle"
          },
          {
             "attribute_code": "price_type",
             "value": "0"
          },
          {
             "attribute_code": "meta_description",
             "value": "New bundle"
          },
          {
             "attribute_code": "tax_class_id",
             "value": "2"
          },
          {
             "attribute_code": "weight_type",
             "value": "0"
          },
          {
             "attribute_code": "category_ids",
             "value": [
                "3",
                "5"
             ]
          },
          {
             "attribute_code": "short_description",
             "value": "<p>New bundle</p>"
          }
       ]
    }
```

## 2. Upload product image for bundle product

Now that we have created a bundle product, we need to upload product image to it.

1. Update the product image by `SKU`.
1. The product image should be `base64encode`.

> 1. Image supported `type` must be `image/png`.

Reviewer: What abut image/jpeg?

Author: sorry I missed that I will again update & push the changes

Author: I have updated & pushed the changes @BarnyShergold

> ---

Reviewer: The word "Cardio" has been misspelt around 40 times across the files, it appears as "Cardo" - please correct all of these.

Author: @BarnyShergold as per suggestion i have updated

layout: tutorial
group: rest-api
title: Step 3. Create the bundle product
subtitle: Create a bundle product tutorial
menu_title: Step 3. Create the bundle product
menu_order: 20
level3_subgroup: bundle-product-tutorial
return_to:
  title: REST tutorials
  url: rest/tutorials/index.html
functional_areas:
  - Integration
contributor_name: Ziffity
contributor_link: https://ziffity.com/
---

We have the information we need to create the Quest Pursuit Jump Cardo Ball Clone bundle product.

The `visibility` attribute is set to 4, meaning the product can be found by browsing or searching.

> The below table shows the equivalent code for the visibility attribute

Reviewer: ```suggestion
The below table shows the equivalent code for the visibility attribute:
```

Author: @BarnyShergold as per suggestion i have updated

                     "can_change_quantity":1
                  }
               ]
            },
            {
               "option_id":3,
               "title":"Cardo Ball",
               "required":true,
               "type":"radio",
               "position":4,
               "sku":"24-WG080",
               "product_links":[
                  {
                     "id":"8",
                     "sku":"24-UG07",
                     "option_id":3,
                     "qty":1,
                     "position":1,
                     "is_default":true,
                     "price":12,
                     "price_type":null,
                     "can_change_quantity":1
                  }
               ]
            }
         ]
      },
      "custom_attributes":[
         {
            "attribute_code":"price_view",
            "value":"0"
         },
         {
            "attribute_code":"meta_title",
            "value":"Quest Pursuit Jump Cardo Ball Kit"
         },
         {
            "attribute_code":"meta_keyword",
            "value":"Quest Pursuit Jump Cardo Ball Kit"
         },
         {
            "attribute_code":"meta_description",

>             "value":" <p>Set of three products like  Quest Lumaflex™ Band,Pursuit Lumaflex™ Tone Band,Zing Jump Rope,Dual Handle Cardio Ball</p>"

Reviewer: ```suggestion
            "value":" <p>Set of three products like Quest Lumaflex™ Band, Pursuit Lumaflex™ Tone Band, Zing Jump Rope, Dual Handle Cardio Ball</p>"
```

Author: @BarnyShergold as per suggestion i have updated

         },
         {
            "attribute_code":"short_description",

>             "value":"<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want  . The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"

Reviewer: ```suggestion
            "value":"<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"
```

Author: @BarnyShergold as per suggestion i have updated

         },
         {
            "attribute_code":"description",

>             "value":"<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want  . The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"

Reviewer: ```suggestion
            "value":"<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"
```

Author: @BarnyShergold as per suggestion i have updated

        },
        {
            "attribute_code": "options_container",
            "value": "container2"
        },
        {
            "attribute_code": "price_view",
            "value": "0"
        },
        {
            "attribute_code": "url_key",
            "value": "quest-pursuit-jump-cardo-ball-kit"
        },
        {
            "attribute_code": "msrp_display_actual_price_type",
            "value": "0"
        },
        {
            "attribute_code": "required_options",
            "value": "0"
        },
        {
            "attribute_code": "has_options",
            "value": "0"
        },
        {
            "attribute_code": "meta_title",
            "value": "Quest Pursuit Jump Cardo Ball Kit"
        },
        {
            "attribute_code": "sku_type",
            "value": "0"
        },
        {
            "attribute_code": "meta_keyword",
            "value": "Quest Pursuit Jump Cardo Ball Kit"
        },
        {
            "attribute_code": "price_type",
            "value": "0"
        },
        {
            "attribute_code": "tax_class_id",
            "value": "2"
        },
        {
            "attribute_code": "meta_description",

>             "value": " <p>Set of three products like  Quest Lumaflex™ Band,Pursuit Lumaflex™ Tone Band,Zing Jump Rope,Dual Handle Cardio Ball</p>"

Reviewer: ```suggestion
            "value": " <p>Set of three products like Quest Lumaflex™ Band, Pursuit Lumaflex™ Tone Band, Zing Jump Rope, Dual Handle Cardio Ball</p>"
```

        },
        {
            "attribute_code": "weight_type",
            "value": "0"
        },
        {
            "attribute_code": "category_ids",
            "value": [
                "3",
                "5"
            ]
        },
        {
            "attribute_code": "short_description",

>             "value": "<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want  . The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"

Reviewer: ```suggestion
            "value": "<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardio ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"
```

Author: @BarnyShergold as per suggestion i have updated

>             "value": "<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardio ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"

Reviewer: ```suggestion
            "value": "<p>The Quest Pursuit Jump Cardio Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardio ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"
```

Author: @BarnyShergold as per suggestion i have updated

        },
        {
            "attribute_code": "description",

>             "value": "<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want  . The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"

Reviewer: ```suggestion
            "value": "<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"
```

Author: @BarnyShergold as per suggestion i have updated

>             "value": "<p>The Quest Pursuit Jump Cardo Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardo ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"

Reviewer: ```suggestion
            "value": "<p>The Quest Pursuit Jump Cardio Ball Kit helps stock your studio with the basics you need for a full-range workout. The kit is composed of four best-selling Luma accessories in one easy bundle: quest, zing jump, dual handle cardio ball. Choose what you want. The kit includes:</p><ul><li>Pursuit Lumaflex™ Tone Band</li><li>Zing Jump Rope</li><li>Dual Handle Cardio Ball</li></ul>"
```

Author: @BarnyShergold as per suggestion i have updated
